src/focus.service.ts

```diff
@@ -416,25 +416,35 @@ export class FocusService {
     const { width, height } = rect;
     let { top, left } = rect;
 
-    for (let parent = el.parentElement; parent !== null; parent = parent.parentElement) {
+    for (let parent = el.parentElement; parent !== container.parentElement; parent = parent.parentElement) {
 
       // Special case: treat the body as the viewport as far as scrolling goes.
-      const prect = parent === document.body
-        ? { top: 0, left: 0, height: window.innerHeight, width: window.innerWidth }
+      const prect = parent === container
+        ? {
+          top: Number(window.getComputedStyle(container, null).paddingTop.slice(0, -2)),
+          left: 0,
+          height: container.clientHeight - Number(window.getComputedStyle(container, null).paddingBottom.slice(0, -2)),
```

Comment:
Is there any way to avoid this? getComputedStyle will trigger a reflow of the entire page.